workflow_array_ephys/analysis.py

```diff
@@ -0,0 +1,118 @@
+class SpikesAlignment(dj.Computed):
+    definition = """
+    -> SpikesAlignmentCondition
+    """
+
+    class AlignedTrialSpikes(dj.Part):
+        definition = """
+        -> master
+        -> ephys.CuratedClustering.Unit
+        -> SpikesAlignmentCondition.Trial
+        ---
+        aligned_spike_times: longblob # (s) spike times relative to alignment event time
+        """
+
+    class UnitPSTH(dj.Part):
+        definition = """
+        -> master
+        -> ephys.CuratedClustering.Unit
+        ---
+        psth: longblob  # event-aligned spike peristimulus time histogram (PSTH)
+        psth_edges: longblob  
+        """
+
+    def make(self, key):
+        unit_keys, unit_spike_times = (ephys.CuratedClustering.Unit & key
+                                       ).fetch('KEY', 'spike_times', order_by='unit')
+        bin_size = (SpikesAlignmentCondition & key).fetch1('bin_size')
+
+        trialized_event_times = trial.get_trialized_alignment_event_times(
+            key, trial.Trial & (SpikesAlignmentCondition.Trial & key))
+
+        min_limit = (trialized_event_times.event - trialized_event_times.start).max()
+        max_limit = (trialized_event_times.end - trialized_event_times.event).max()
+
+        # Spike raster
+        aligned_trial_spikes = []
+        units_spike_raster = {u['unit']: {**key, **u, 'aligned_spikes': []
+                                          } for u in unit_keys}
+        for _, r in trialized_event_times.iterrows():
+            if np.isnan(r.event):
+                continue
+            alignment_start_time = r.event - min_limit
+            alignment_end_time = r.event + max_limit
+            for unit_key, spikes in zip(unit_keys, unit_spike_times):
+                aligned_spikes = spikes[(alignment_start_time <= spikes)
+                                        & (spikes < alignment_end_time)] - r.event
+                aligned_trial_spikes.append({**key, **unit_key,
+                                             **r.trial_key,
+                                             'aligned_spike_times': aligned_spikes})
+                units_spike_raster[unit_key['unit']]['aligned_spikes'
+                                                     ].append(aligned_spikes)
+
+        # PSTH
+        for unit_spike_raster in units_spike_raster.values():
+            spikes = np.concatenate(unit_spike_raster['aligned_spikes'])
+
+            psth, edges = np.histogram(spikes,
+                                       bins=np.arange(-min_limit, max_limit, bin_size))
+            unit_spike_raster['psth'] = (psth
+                                         / len(unit_spike_raster.pop('aligned_spikes'))
+                                         / bin_size)
+            unit_spike_raster['psth_edges'] = edges[1:]
+
+        self.insert1(key)
+        self.AlignedTrialSpikes.insert(aligned_trial_spikes)
+        self.UnitPSTH.insert(list(units_spike_raster.values()))
+
+    def plot_raster(self, key, unit, axs=None):
```

Comment:
```suggestion
    def plot(self, key, unit, axs=None):
```

Reply:
Since this creates both the raster and PSTH plots.  If we make this change, we will have to update the Jupyter notebook.

Reply:
Since we are calling it as follows `analysis.SpikesAlignment().plot_raster()` I don't think this should be an issue.